core/src/main/java/com/netflix/iceberg/TableProperties.java

```diff
@@ -66,4 +66,8 @@
   public static final boolean OBJECT_STORE_ENABLED_DEFAULT = false;
 
   public static final String OBJECT_STORE_PATH = "write.object-storage.path";
+
+  // This only applies to files written after this property is set. Files previously written aren't relocated to
+  // reflect this parameter.
+  public static final String WRITE_NEW_DATA_LOCATION = "write.data.location";
```

Comment:
Should this property mirror the existing `write.object-storage.path` property? Maybe `write.folder-storage.path` would be better. Also, I would like to see a comment about what happens when this isn't set. That behavior should be to default to a `data` folder under the table location.